tensorflow_similarity/losses/vicreg.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Callable, Dict, Optional
```

Comment:
Vicreg is already in development and this version seems to cause a conflict. Should we remove this from this PR?

Reply:
I believe i changed some important things - previous version had some small issues regarding tf ops and customizability. This should replace previous vicreg - have confirmed this version works pretty well.

Reply:
Sounds good. I went ahead and resolved the merge conflicts by taking all the changes from this PR.